public/scripts/report.js

```diff
@@ -26,12 +28,47 @@
   function storePosition(position) {
     var coords = position.coords.latitude + ',' + position.coords.longitude;
 
-    locationButton.setAttribute('value', coords);
-    locationButton.innerHTML = 'Using current location';
-    locationButton.classList.remove('getting-location');
-    locationButton.classList.add('using-location');
+    useLocationButton.setAttribute('value', coords);
+    useLocationButton.innerHTML = 'Using current location';
+    useLocationButton.classList.remove('getting-location');
+    useLocationButton.classList.add('using-location');
+    searchLocationButton.innerHTML = 'Search Location';
+    searchLocationButton.classList.remove('using-location');
+
   };
 
+  // SELECT LOCATION:
+
+  var closeButton = document.getElementById('closeButton');
+
+  // close map view onclick
+  closeButton.addEventListener('click', function(event){
```

Comment:
here you've written the callback as an anonymous function (i.e. the second argument of addEventListener) rather than a separate function